### PR TITLE
Add assigned routes

### PR DESCRIPTION
**Pull request by skeltont**
Adds the ability to assign routes to users of user_level 'Setter' or above. builds off of my previous pull request (for issue #20) in response to issue #22.

### CODE REVIEW

File: app/controllers/assigned_route_controller.rb [hunk 1]
```diff
@@ -1,10 +1,54 @@
...
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
   def show
   end
 
   def index
+    @assigned_route = AssignedRoute.new
+    @routes = AssignedRoute.joins(:user)
+    @setters = User.where('user_level > 0').order(:fname)
```
Comment (jonahgeorge) [inline app/controllers/assigned_route_controller.rb]: Move this into a named scope

File: app/controllers/assigned_route_controller.rb [hunk 2]
```diff
@@ -1,10 +1,54 @@
...
+    @setters.each do |setter|
+      setter.full_name = setter.fname + " " + setter.lname
```
Comment (jonahgeorge) [inline app/controllers/assigned_route_controller.rb]: Use the `User#name` method instead (described below)

File: app/controllers/user_controller.rb [hunk 3]
```diff
@@ -11,6 +13,7 @@ def index
 
   def show
     @user = User.find(params[:id])
+    @assigned_routes = AssignedRoute.where(:user_id => params[:id], :completed => 0, :hidden => 0)
```
Comment (jonahgeorge) [inline app/controllers/user_controller.rb]: Use a named scope

File: app/models/user.rb [hunk 4]
```diff
@@ -13,6 +13,8 @@ class User < ActiveRecord::Base
   has_many :routes, inverse_of: :user
   has_many :support_tickets, inverse_of: :user
 
+  attr_accessor :full_name
```
Comment (jonahgeorge) [inline app/models/user.rb]: Make this

``` ruby
def name
  "#{fname} #{lname}"
end
```

**Comment by jonahgeorge** [comment]
Added some goodies for you.

**Comment by skeltont** [comment]
Thanks for the tips there, I'll fix those here in the next commit. And yeah I believe this pr encapsulates the commit for the one before as well. 

Still getting used to rails.

File: app/controllers/assigned_route_controller.rb [hunk 5]
```diff
@@ -1,10 +1,54 @@
...
+    end
+    @gyms = {
+      "McAlexander" => "McAlexander",
+      "Dixon"       => "Dixon"
+    }
+    @grades = {
```
Comment (jonahgeorge) [inline app/controllers/assigned_route_controller.rb:18]: Let's add a `TODO` comment on these just as a reminder

**Comment by skeltont** [comment]
Made the commit above in response to your comments @jonahgeorge

**Comment by jonahgeorge** [comment]
Wrap the Reports dropdown with a `<% if user_signed_in? %>` because `assigned_route_index_path` expects `current_user` to not be nil for the policy check. If you navigate to the route without logging in, it throws an error.

``` ruby
<% if user_signed_in? %>
  <li class="dropdown">
    <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Reports <span class="caret"></span></a>
    <ul class="dropdown-menu">
      <li><%= link_to "Assigned Routes", assigned_route_index_path %></li>
    </ul>
  </li>
<% end %>
```

**Comment by skeltont** [comment]
We might want to do the same thing with User tab in the future too. I have a lot of thoughts on that actually, but they're not mission critical atm.

**Pull request merged**